### PR TITLE
Switch boost::shared_ptr to urdf::*SharedPtr

### PR DESCRIPTION
**Pull request by bhomberg**
This change was required for me to build on melodic.

### CODE REVIEW

**Comment by gavanderhoorn** [comment]
This would seem to be https://github.com/ros-industrial/fanuc/pull/262, but for the packages in this repository.

File: fanuc_lrmate200id_moveit_plugins/CMakeLists.txt [hunk 1]
```diff
@@ -24,6 +26,8 @@ catkin_package(
     pluginlib
     roscpp
     tf_conversions
+  DEPENDS
+    urdfdom_headers
```
Comment (gavanderhoorn) [inline fanuc_lrmate200id_moveit_plugins/CMakeLists.txt:30]: Is this really necessary? Doesn't the `depend` on `moveit_core` already pull this in ([here](https://github.com/ros-planning/moveit/blob/e6d0f3442c39e2b6948f443ccd2e93be190c4a61/moveit_core/package.xml#L33))?

Similar comment for the other `urdfdom_headers` related statements.

File: fanuc_lrmate200id_moveit_plugins/lrmate200id_kinematics/src/fanuc_lrmate200id_manipulator_ikfast_moveit_plugin.cpp [hunk 2]
```diff
@@ -299,12 +299,12 @@ bool IKFastKinematicsPlugin::initialize(const std::string &robot_description,
 
   ROS_DEBUG_STREAM_NAMED("ikfast","Reading joints and links from URDF");
 
-  boost::shared_ptr<urdf::Link> link = boost::const_pointer_cast<urdf::Link>(robot_model.getLink(tip_frame_));
+  urdf::LinkSharedPtr link = urdf::const_pointer_cast<urdf::Link>(robot_model.getLink(tip_frame_));
```
Comment (gavanderhoorn) [inline fanuc_lrmate200id_moveit_plugins/lrmate200id_kinematics/src/fanuc_lrmate200id_manipulator_ikfast_moveit_plugin.cpp:302]: In ros-industrial/fanuc#262 the `Const` variants of the `*SharedPtr`s were used where appropriate ([here](https://github.com/ros-industrial/fanuc/blob/75a6f7045c87e3c4394407b39a3bdf7b8902833f/fanuc_lrmate200i_moveit_plugins/lrmate200i_kinematics/src/fanuc_lrmate200i_manipulator_ikfast_moveit_plugin.cpp#L302) fi).

Would be nice to do that here as well.

**Comment by gavanderhoorn** [comment]
Tbh I'm not sure right now why Travis fails to build this:

```
/root/catkin_ws/src/fanuc_experimental/fanuc_lrmate200id_moveit_plugins/lrmate200id_kinematics/src/fanuc_lrmate200id_manipulator_ikfast_moveit_plugin.cpp:302:30: error: ‘const_pointer_cast’ is not a member of ‘urdf’

   urdf::LinkSharedPtr link = urdf::const_pointer_cast<urdf::Link>(robot_model.getLink(tip_frame_));

                              ^
```

It builds on my machine :man_shrugging:

**Comment by gavanderhoorn** [comment]
Apologies @bhomberg, I just realised I've essentially broke this PR (seeing as the files that it attempts to change are now in `ros-industrial/fanuc`).

I'll port your changes there.

**Comment by gavanderhoorn** [comment]
I've applied the fix from ros-industrial/fanuc#262 to the LR Mate 200iD's IKFast plugin package, which should resolve the issue on Melodic for you @bhomberg.

See fd000a8a91a503113ad5a4e5c54c338a38bae78f.

**Comment by gavanderhoorn** [comment]
Closing this as it's no longer needed.

Thanks for bringing this to my attention @bhomberg

**Comment by bhomberg** [comment]
Thanks for fixing, @gavanderhoorn !

**Pull request closed without merging**